ods_ci/tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot

```diff
@@ -92,26 +98,40 @@ Select Container Size
     Click Element  xpath:${JUPYTERHUB_DROPDOWN_XPATH}\[1]
     Click Element  xpath://span[.="${container_size}"]/../..
 
-Wait Until GPU Dropdown Exists
-    [Documentation]    Verifies that the dropdown to select the no. of GPUs exists
-    Wait Until Page Contains    Number of GPUs
+Wait Until Accelerator Dropdown Exists
+    [Documentation]    Verifies that the dropdown to select the Accelerator exists
     Page Should Not Contain    All GPUs are currently in use, try again later.
-    Wait Until Page Contains Element    xpath:${KFNBC_GPU_DROPDOWN_XPATH}
-    ...    error=GPU selector is not present in JupyterHub Spawner
+    Wait Until Page Contains Element    xpath:${KFNBC_ACCELERATOR_DROPDOWN_XPATH}
+    ...    error=Accelerator selector is not present in JupyterHub Spawner
 
-Set Number Of Required GPUs
-    [Documentation]  Sets the gpu count based on the ${gpus} argument
+Set NVidia GPU Accelerator
+    [Documentation]  Set NVidia GPU Accelerator
+    Click Element  xpath:${KFNBC_ACCELERATOR_DROPDOWN_XPATH}
+    Click Element  xpath:${KFNBC_ACCELERATOR_DROPDOWN_NVIDIA_XPATH}
+
+Set Number Of Required Accelerators
+    [Documentation]  Sets the Accelerators count based on the ${gpus} argument
     [Arguments]  ${gpus}
-    Click Element  xpath:${KFNBC_GPU_DROPDOWN_XPATH}
-    Click Element  xpath:${KFNBC_GPU_DROPDOWN_XPATH}/../..//button[.="${gpus}"]
+    ${acc_num} =   Get Value   xpath:${KFNBC_ACCELERATOR_INPUT_XPATH}
+    Log    Actual num of Accelerators: ${acc_num}
+    IF    ${acc_num} != ${gpus}
+        Input Text  ${KFNBC_ACCELERATOR_INPUT_XPATH}  ${gpus}
+    END
+
 
 Fetch Max Number Of GPUs In Spawner Page
     [Documentation]    Returns the maximum number of GPUs a user can request from the spawner
-    ${gpu_visible} =    Run Keyword And Return Status    Wait Until GPU Dropdown Exists
+    ${gpu_visible} =    Run Keyword And Return Status    Wait Until Accelerator Dropdown Exists
     IF  ${gpu_visible}==True
-       Click Element    xpath:${KFNBC_GPU_DROPDOWN_XPATH}
-       ${maxGPUs} =    Get Text    xpath://li[@class="pf-c-select__menu-wrapper"][last()]/button
-       ${maxGPUs} =    Convert To Integer    ${maxGPUs}
+       Set NVidia GPU Accelerator
+       ${max_operator_detected} =  Run Keyword And Return Status    Page Should Contain Element    xpath=${KFNBC_MAX_ACCELERATOR_WARNING_XPATH}
```

Comment:
is this applicable for any accelerator or GPU only?

Reply:
No. It was NVidia specific. I did not wanna to overengineer things as we are testing only with NVidia rn.
But it also makes sense to be generic and Accelerator Technology dependant.
Fixed in [cbf6eec](https://github.com/red-hat-data-services/ods-ci/pull/971/commits/cbf6eec202c7ac338bdaeea757e0597f81b7df61)